esp/services/esdl_svc_engine/esdl_binding.cpp

```diff
@@ -67,75 +67,6 @@ bool trimXPathToParentSDSElement(const char *element, const char * xpath, String
-    VStringBuffer xpath("%s[@id='%s'][1]", ESDL_DEF_ENTRY, id);
-
-    ESPLOG(LogNormal, "ESDL Binding: Fetching ESDL Definition from Dali: %s ", id);
-
-    Owned<IRemoteConnection> conn = querySDS().connect(ESDL_DEFS_ROOT_PATH, myProcessSession(), RTM_LOCK_READ, SDS_LOCK_TIMEOUT_DESDL);
-    if (!conn)
-       throw MakeStringException(-1, "Unable to connect to ESDL Service definition information in dali '%s'", ESDL_DEFS_ROOT_PATH);
-
-    IPropertyTree * esdlDefinitions = conn->queryRoot();
-
-    toXML(esdlDefinitions->queryPropTree(xpath), exsdl);
-}
-
-void fetchESDLDefinitionFromDali(const char *esdlServiceName, unsigned targetVersion, StringBuffer & exsdl)
-{
-    if (!esdlServiceName || !*esdlServiceName)
-        throw MakeStringException(-1, "Unable to fetch ESDL Service definition information, service name is not available");
-    if (targetVersion == 0 )
-        throw MakeStringException(-1, "Unable to fetch ESDL Service definition information, target version number is invalid: '%s.%d", esdlServiceName, targetVersion);
-
-    DBGLOG("ESDL Binding: Fetching ESDL Definition from Dali: %s.%d ", esdlServiceName, targetVersion);
-
-    StringBuffer id(esdlServiceName);
-    id.toLowerCase().append(".").append(targetVersion);
-
-    return fetchESDLDefinitionFromDaliById(id, exsdl);
-}
-
-IPropertyTree * fetchESDLBindingFromDali(const char *process, const char *bindingName)
-{
-    try
-    {
-        //There shouldn't be multiple entries here, but if so, we'll use the first one
-        VStringBuffer xpath("%s[@id='%s.%s'][1]", ESDL_BINDING_ENTRY, process, bindingName);
-
-        Owned<IRemoteConnection> conn = querySDS().connect(ESDL_BINDINGS_ROOT_PATH, myProcessSession(), RTM_LOCK_READ, SDS_LOCK_TIMEOUT_DESDL);
-        if (!conn)
-        {
-            ESPLOG(LogMin, "Unable to connect to ESDL Service binding information in dali %s", ESDL_BINDINGS_ROOT_PATH);
-            return nullptr;
-        }
-
-        ESPLOG(LogNormal, "ESDL Binding: Fetching ESDL Binding from Dali %s[@EspProcess='%s'][@EspBinding='%s'][1]", ESDL_BINDING_ENTRY, process, bindingName);
-
-        if (conn->queryRoot()->hasProp(xpath))
-            return createPTreeFromIPT(conn->queryRoot()->queryPropTree(xpath));
-        else
-            return nullptr;
-    }
-    catch (IException *E)
-    {
-        VStringBuffer message("ESDL Binding: Error fetching ESDL Binding %s[@EspProcess='%s'][@EspBinding='%s'][1] from Dali.", ESDL_BINDING_ENTRY, process, bindingName);
-        EXCLOG(E, message);
-        E->Release();
-    }
-    catch(...)
-    {
-        ESPLOG(LogMin, "ESDL Binding: Unknown error encountered while fetching ESDL Binding from Dali %s[@EspProcess='%s'][@EspBinding='%s'][1]", ESDL_BINDING_ENTRY, process, bindingName);
-    }
-
-    return NULL;
-}
-
 IPropertyTree * fetchESDLBindingFromStateFile(const char *process, const char *bindingName, const char * stateFileName)
```

Comment:
why don't we move this method to your new class as well?

Reply:
For this pass I'd like to leave the state file out and only deal with "central" repository.